include/zephyr/devicetree.h

```diff
@@ -4066,6 +4102,19 @@
 #define DT_INST_CHILD(inst, child) \
 	DT_CHILD(DT_DRV_INST(inst), child)
 
+/**
+ * @brief Get a node identifier for a child node by a unit address of DT_DRV_INST(inst)
+ *
+ * @param inst instance number
+ * @param addr Unit address for the child node. Can be hexadecimal (prefix with 0x) or decimal
```

Comment:
Either "see `DT_CHILD_BY_UNIT_ADDR` for restrictions" or:

```suggestion
 * @param addr Unit address for the child node. Can be decimal or
 *             hexadecimal (lowercase and prefixed with 0x)
```